src/librustc_typeck/check/method/suggest.rs

```diff
@@ -271,13 +277,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                             "you must specify a type for this binding, like `{}`",
                                             concrete_type,
                                         );
-                                        match parent_node {
-                                            hir_map::NodeLocal(hir::Local {
+
+                                        match (is_real_filename, parent_node) {
```

Comment:
You match on filename above toget this bool. Can you just match on it directly here?

Reply:
Done.